lauth/spec/requests/authorized_client_ip_spec.rb

```diff
@@ -0,0 +1,88 @@
+RSpec.describe "/authorized by client-ip", type: [:request, :database] do
+  # Convenience for building networks.
+  # Requires 'institution' be set in a let block.
+  # @param access [String] Either 'allow' or 'deny'
+  # @param cidr [String] A IPv4 CIDR range. We attempt to mirror those ranges that
+  #   are in db/network.sql for clarity.
+  def build_network(access, cidr)
+    Factory[
+      :network, :for_institution, institution: institution,
+      dlpsAccessSwitch: access, dlpsCIDRAddress: cidr
+    ]
+  end
+
+  # Request the location from the provided ip address
+  # @param ip [String]
+  # @return [Hash] the response body after json parsing
+  def request_from(ip)
+    get "/authorized", {user: "", uri: "/restricted-by-client-ip", ip: ip}
+    JSON.parse(last_response.body, symbolize_names: true)
+  end
+
+  let!(:institution) { Factory[:institution] }
+  let!(:collection) { Factory[:collection, :restricted_by_client_ip] }
+  let!(:grant) { Factory[:grant, :for_institution, institution: institution, collection: collection] }
+
+  context "(allow,null) given an allowed network only" do
+    let!(:network) { build_network("allow", "10.1.16.0/24") }
+
+    it "is allowed within the network" do
+      expect(request_from("10.1.16.2")).to eq({determination: "allowed"})
+    end
+
+    it "is denied outside the network" do
+      expect(request_from("10.1.17.1")).to eq({determination: "denied"})
+    end
+  end
+
+  context "(allow,allow) given an allowed enclave within an allowed network" do
+    let!(:network) { build_network("allow", "10.1.6.0/24") }
+    let!(:enclave) { build_network("allow", "10.1.6.8/29") }
```

Comment:
Revisit the specifics/range here -- I have to play computer too much to check this for my taste.